### PR TITLE
Upgrade to Knackpy v1

### LINKED ISSUE #4156: Upgrade atd-service-bot to Knackpy v1.0

### PR DESCRIPTION
**Pull request by johnclary**
Fixes [#4156](https://github.com/cityofaustin/atd-data-tech/issues/4156)

- upgrades to knackpy v1.0
- cleans up the issue formatting logic thanks to Knackpy 1.0
- removes some crufty `for` loops that were remnants of the pre-monorepo days
- switches to use of environmental variables instead of secrets.py
- removes argutil dependency
- pins requirements.txt packages to minor release
- updated the `from_csv.py` helper script to use an access token instead of (now deprecated) basic auth

### CODE REVIEW

File: intake.py [hunk 1]
```diff
@@ -1,113 +1,80 @@
...
     for field in fields:
         """ Formatting and placement of Knack issue text is driven by
         config/config.py
         """
-        knack_field_label = knack_field_map[field["knack"]]["label"]
+        knack_field_id = field["knack"]
+        knack_field_label = issue.fields[knack_field_id].name
+        knack_field_value = issue.fields[knack_field_id].formatted
+
+        if not knack_field_value:
```
Comment (mddilley) [inline intake.py:50]: Handling the fields looks so clean now. Nice!

File: intake.py [hunk 2]
```diff
@@ -249,89 +165,79 @@ def form_submit(token, app_id, scene, view, payload):
...
 
-    env = args.env
+    view = KNACK_APP["api_view"]["view"]
+    app = knackpy.App(app_id=KNACK_APP_ID, api_key=KNACK_API_KEY)
 
-    KNACK_USERNAME = KNACK_CREDS[env].get("username")
-    KNACK_PASSWORD = KNACK_CREDS[env].get("password")
-    KNACK_API_KEY = KNACK_CREDS[env].get("api_key")
-    KNACK_APP_ID = KNACK_CREDS[env].get("app_id")
+    issues = app.get(view)
```
Comment (mddilley) [inline intake.py:183]: Knackpy v1 makes getting the issue records look easy 🙌

**Review by mateoclarke — APPROVED** [review]
👍

**Review by mddilley — APPROVED** [review]
Awesome to see Knackpy v1 and the monorepo make this code simpler! Very cool that this is now running in Airflow as well. 👏

**Comment by johnclary** [comment]
thanks, y'all!

**Pull request merged**